diff --git a/src/main.cpp b/src/main.cpp
index 7f02ab7..b58f147 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -332,6 +332,7 @@ int main(int argc, char** argv) {
     program_argv0 = argv[0];
 
     QApplication app(argc, argv);
+    QIcon::setThemeName(QStringLiteral("breeze"));
     app.setApplicationVersion(QStringLiteral(LXQT_ARCHIVER_VERSION));
     app.setQuitOnLastWindowClosed(true);
     app.setAttribute(Qt::AA_UseHighDpiPixmaps);
