From ab5d3a0ecd1c4854d0b6f77e977acc92bf129e19 Mon Sep 17 00:00:00 2001
From: Zamarin Arthur <arthurzam@gmail.com>
Date: Thu, 2 Nov 2017 07:03:41 +0200
Subject: [PATCH] Fix utemper library search (#148)

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index edb70a1..bd8a663 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,6 +114,7 @@ set(QTERMWIDGET_INCLUDE_DIR "${CMAKE_INSTALL_FULL_INCLUDEDIR}/${QTERMWIDGET_LIBR
 
 CHECK_FUNCTION_EXISTS(updwtmpx HAVE_UPDWTMPX)
 CHECK_INCLUDE_FILE(utempter.h HAVE_UTEMPTER)
+FIND_LIBRARY(HAVE_UTEMPTER_LIB NAMES ulog utempter)
 
 qt5_wrap_cpp(MOCS ${HDRS})
 qt5_wrap_ui(UI_SRCS ${UI})
@@ -162,12 +163,12 @@ if(HAVE_UPDWTMPX)
     )
 endif()
 
-if(HAVE_UTEMPTER)
+if(HAVE_UTEMPTER AND HAVE_UTEMPTER_LIB)
     target_compile_definitions(${QTERMWIDGET_LIBRARY_NAME}
         PRIVATE
             "HAVE_UTEMPTER"
     )
-    target_link_libraries(${QTERMWIDGET_LIBRARY_NAME} ulog)
+    target_link_libraries(${QTERMWIDGET_LIBRARY_NAME} ${HAVE_UTEMPTER_LIB})
 endif()
 
 if (UTF8PROC_FOUND)
