diff --git a/sudo.cpp b/sudo.cpp
index 6a16314..7ab843a 100644
--- a/sudo.cpp
+++ b/sudo.cpp
@@ -251,23 +251,12 @@ void Sudo::child()
     std::string program = backendName().toLocal8Bit().data();
 
     std::string preserve_env_param;
-    switch (mBackend)
     {
-        case BACK_SUDO:
-            preserve_env_param = "--preserve-env=";
-
-            preserve_env_param += env_workarounds();
-
-            *(param_arg++) = preserve_env_param.c_str(); //preserve environment
+            *(param_arg++) = "-E";
             *(param_arg++) = "/bin/sh";
-            break;
-        case BACK_SU:
-        case BACK_NONE:
-            env_workarounds();
-            break;
-
     }
     *(param_arg++) = "-c"; //run command
+    unsetenv("DBUS_SESSION_BUS_ADDRESS");
 
     params[0] = program.c_str();
 
