diff --git a/sudo.cpp b/sudo.cpp
index 2282512..6242c9a 100644
--- a/sudo.cpp
+++ b/sudo.cpp
@@ -83,7 +83,7 @@ namespace
         QTextStream(stdout)
             << QObject::tr("%1 version %2\n").arg(app_master).arg(app_version);
     }
-
+/*
     //Note: array must be sorted to allow usage of binary search
     static constexpr char const * const ALLOWED_VARS[] = {
         "DISPLAY"
@@ -100,11 +100,11 @@ namespace
                         , [] (char const * const a, char const * const b) { return strcmp(a, b) < 0; }));
         }
     };
-    assert_helper h;
+	assert_helper h;*/
 
     inline void env_workarounds()
     {
-        std::cerr << LXQTSUDO << ": Stripping child environment except for: ";
+		/*std::cerr << LXQTSUDO << ": Stripping child environment except for: ";
         std::copy(ALLOWED_VARS, ALLOWED_END - 1, std::ostream_iterator<const char *>{std::cerr, ", "});
         std::cerr << *(ALLOWED_END - 1) << '\n'; // printing the last separately to avoid trailing comma
         // cleanup environment, because e.g.:
@@ -119,7 +119,10 @@ namespace
             {
                 unsetenv(key.toStdString().c_str());
             }
-        }
+		}*/
+		//cleanup environment
+		//pcmanfm-qt will not start if the DBUS_SESSION_BUS_ADDRESS is preserved
+		unsetenv("DBUS_SESSION_BUS_ADDRESS");
     }
 }
 
