--- a/drivers/cpufreq/cpufreq_performance.c	2021-04-25 23:49:08.000000000 +0300
+++ b/drivers/cpufreq/cpufreq_performance.c	2021-05-08 09:31:14.524433090 +0300
@@ -24,12 +24,10 @@
 	.limits		= cpufreq_gov_performance_limits,
 };
 
-#ifdef CONFIG_CPU_FREQ_DEFAULT_GOV_PERFORMANCE
 struct cpufreq_governor *cpufreq_default_governor(void)
 {
 	return &cpufreq_gov_performance;
 }
-#endif
 #ifndef CONFIG_CPU_FREQ_GOV_PERFORMANCE_MODULE
 struct cpufreq_governor *cpufreq_fallback_governor(void)
 {
