diff --git a/src/qmplay2/YouTubeDL.cpp b/src/qmplay2/YouTubeDL.cpp
index f05e615b..39039ecf 100644
--- a/src/qmplay2/YouTubeDL.cpp
+++ b/src/qmplay2/YouTubeDL.cpp
@@ -37,11 +37,7 @@ static QMutex g_mutex(QMutex::Recursive);
 
 QString YouTubeDL::getFilePath()
 {
-    return QMPlay2Core.getSettingsDir() + "youtube-dl"
-#ifdef Q_OS_WIN
-    ".exe"
-#endif
-    ;
+    return QStringLiteral("/usr/bin/youtube-dl");
 }
 QStringList YouTubeDL::getCommonArgs()
 {
@@ -357,6 +353,7 @@ bool YouTubeDL::download()
 }
 bool YouTubeDL::update()
 {
+#if 0
     // Mutex must be locked here
 
     qDebug() << "\"youtube-dl\" updates will be checked";
@@ -421,6 +418,7 @@ bool YouTubeDL::update()
     }
 
     QMPlay2Core.setWorking(false);
+#endif
     return true;
 }
 
