diff --git a/src/qmplay2/YouTubeDL.cpp b/src/qmplay2/YouTubeDL.cpp
index d2c390e8..ff02ec0b 100644
--- a/src/qmplay2/YouTubeDL.cpp
+++ b/src/qmplay2/YouTubeDL.cpp
@@ -36,11 +36,7 @@ static QMutex g_mutex(QMutex::Recursive);
 
 QString YouTubeDL::getFilePath()
 {
-    return QMPlay2Core.getSettingsDir() + "youtube-dl"
-#ifdef Q_OS_WIN
-    ".exe"
-#endif
-    ;
+    return QStringLiteral("/usr/bin/youtube-dl");
 }
 QStringList YouTubeDL::getCommonArgs()
 {
